src/com/goide/codeInsight/imports/GoOptimizeImportsPass.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Mihai Toader, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.codeInsight.imports;
+
+import com.intellij.codeInsight.daemon.impl.DefaultHighlightInfoProcessor;
+import com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.DocumentUtil;
+import org.jetbrains.annotations.NotNull;
+
+public class GoOptimizeImportsPass extends ProgressableTextEditorHighlightingPass {
+  @NotNull private final PsiFile myFile;
+
+  public GoOptimizeImportsPass(@NotNull Project project, @NotNull PsiFile file, @NotNull Editor editor) {
+    super(project, editor.getDocument(), "Go Optimize Imports Pass", file, editor, file.getTextRange(), false,
+          new DefaultHighlightInfoProcessor());
+    myFile = file;
+  }
+
+  @Override
+  protected void collectInformationWithProgress(@NotNull ProgressIndicator progress) {
+    progress.checkCanceled();
+  }
+
+  @Override
+  protected void applyInformationWithProgress() {
+    final Runnable runnable = new GoImportOptimizer().processFile(myFile);
```

Comment:
So are you going to calculate all imports to remove right in EDT? Don't you think it's not really efficient?